keys.go

```diff
@@ -133,22 +133,122 @@ type KeyVersion struct {
 	CreationDate *time.Time `json:"creationDate,omitempty"`
 }
 
+// This function returns a string so we can pass extra info not in the key struct if needed
+type CreateKeyOption func(k *Key) string
+
+func WithExpiration(expiration *time.Time) CreateKeyOption {
+	return func(key *Key) string {
+		key.Expiration = expiration
+		return "expiration"
+	}
+}
+
+func WithDescription(description string) CreateKeyOption {
+	return func(key *Key) string {
+		key.Description = description
+		return "description"
+	}
+}
+
+func WithPayload(payload, encryptedNonce, iv string, sha1 bool) CreateKeyOption {
+	return func(key *Key) string {
+		ret := "payload"
+		key.Payload = payload
+		if !key.Extractable {
+			algorithm := AlgorithmRSAOAEP256
+			if sha1 {
+				algorithm = AlgorithmRSAOAEP1
+			}
+			key.EncryptedNonce = encryptedNonce
+			key.IV = iv
+			key.EncryptionAlgorithm = algorithm
+			ret += "-" + algorithm
+		}
+		return ret
+	}
+}
+
+func WithAliases(aliases []string) CreateKeyOption {
+	return func(key *Key) string {
+		key.Aliases = aliases
+		return "aliases"
+	}
+}
+
+func WithPolicy(policy *Policy) CreateKeyOption {
+	return func(key *Key) string {
+		/*
+			Setting the value of rotationInterval to -1 in case user passes 0 value
+			as we want to retain the param `interval_month` after marshalling
+			so that we can get correct error msg from REST API saying interval_month should be between 1 to 12
+			Otherwise the param would not be sent to REST API in case of value 0
+			and it would throw error saying interval_month is missing
+		*/
+		if policy.Rotation != nil && policy.Rotation.Interval == 0 {
+			policy.Rotation.Interval = -1
+		}
+		key.Rotation = policy.Rotation
+		key.DualAuthDelete = policy.DualAuth
+		return "policy"
+		/*
+			I needed these CreateKeyOption functions to return a string to ensure I knew when a policy was passed,
+			so that it would call the create key with policy overrides function,
+			even when a policy with empty/nil fields is passed.
+		*/
+	}
+}
+
+func WithTags(tags []string) CreateKeyOption {
+	return func(key *Key) string {
+		key.Tags = tags
+		return "tags"
+	}
+}
+
+func (c *Client) CreateKeyWithOptions(ctx context.Context, name string, extractable bool, options ...CreateKeyOption) (*Key, error) {
+	key := &Key{
+		Name:        name,
+		Type:        keyType,
+		Extractable: extractable,
+	}
+	overrides := false
+	for _, opt := range options {
+		if opt != nil {
+			optName := opt(key)
+			if optName == "policy" {
```

Comment:
Recommend separating into `CreateKeyWithOptions` and `CreateKeyWithPolicies` explicitly because these would call 2 different APIs with 2 different actions that a user can be IAM authorized on. 
Recommend removing policies from `CreateKeyOption` type and have as explicit input `Policy` (user can choose to pass empty policy)